### PR TITLE
feat: implement button component

### PR DESCRIPTION
**Pull request by Macdeal**
Closes - #32

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import classNames from 'classnames';
+import { ReactNode } from 'react';
+
+export enum ButtonTypes {
+  Primary = 'primary',
+  HasIcon = 'has-icon',
+}
+
+interface ButtonProps {
```
Comment (yoki1234) [inline src/components/Button/Button.tsx:9]: is everyThing optional props,
if i am re-calling it , `text` and `onClick` are a mandatory prop.

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+const Button = ({
+  icon,
+  onClick,
+  text,
+  variant,
+  width = 'w-full',
+}: ButtonProps): JSX.Element => {
+  const btnClass = classNames(
+    'rounded-lg flex align-center gap-3 px-[1rem] py-[.75rem] border-dark-orange font-semibold text-white text-center',
```
Comment (yoki1234) [inline src/components/Button/Button.tsx:25]: use provided tailwind measurements `px-[1rem]` to `px-4`, and `py-[.75rem]` to `py-3`

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    {
+      'bg-transparent focus-within:border': variant === ButtonTypes.HasIcon,
+      'bg-dark-orange': variant === ButtonTypes.Primary,
+      [width]: true,
```
Comment (yoki1234) [inline src/components/Button/Button.tsx:29]: please do explain , why we need a style option that is always triggered?, thence it is true.

File: src/container/EventCard.tsx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import EventTitle from '../components/eventTitle/EventTitle';
+import EventTitle from '../components/EventTitle/EventTitle';
```
Comment (yoki1234) [inline src/container/EventCard.tsx:1]: do revert this change

File: src/components/Button/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      {text}
+      {icon && <span>{icon}</span>}
+    </button>
+  );
+};
+
+Button.defaultProps = {
+  onClick: () => {},
+  variant: ButtonTypes.Primary,
+  width: 'w-full',
```
Comment (yoki1234) [inline src/components/Button/Button.tsx:44]: why default prop is `w-full`?

**Review by yoki1234 — CHANGES_REQUESTED** [review]

**Comment by yoki1234** [comment]
Please do Attach image of what you have developed.
as `professionalism brings perfect cuts for making your future sharp.`

File: src/container/EventCard.tsx (hunk 4 above)
Comment (Macdeal) [reply src/container/EventCard.tsx:1]: for what

File: src/components/Button/Button.tsx (hunk 5 above)
Comment (Macdeal) [reply src/components/Button/Button.tsx:44]: then what would be the default value

**Pull request merged**